Scripts/Editor/Compiler/ScriptDefinesToggle.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Build;
+
+namespace Anvil.Unity.DOTS.Editor.Compiler
+{
+    /// <summary>
+    /// Provides a quick and easy way to enable/disable commonly used Scripting Defines for
+    /// development and debugging. 
+    /// </summary>
+    [InitializeOnLoad]
+    public static class ScriptDefinesToggle
```

Comment:
Thinking this should be DEBUG specific rather than a central place for all defines.
Same goes for the `Compiler` namespace probably belongs under `Debug` or core instead.